### ISSUE TITLE
imported 'enumValues' get prefixed when put in input fields

### ISSUE THREAD

**Issue by roseline124** [dotansimha/graphql-code-generator#2976]
**Describe the bug**

When I set typePrefix(ex. 'I'), enums(ex. 'MyEnum') registered as 'enumValues' in codegen.yml are imported in generated file without the prefix (-> 'MyEnum'). 

If I put those enum types in input fields, then those get prefixed in input fields when generated(ex. `myEnum: IMyEnum`). In this case, the enum type name 'IMyEnum' in generated input fields is different with the enum type name 'Enum', so the enum type evokes Not Found Error.

If you'd like to fix this issue, it helps for you to look `packages/plugins/other/visitor-plugin-common/src/variables-to-object.ts` I guess.

(+ `enumPrefix: false` is not a solution for this problem, it throws another error..)

**To Reproduce**

1. codegen.yml

```yml
overwrite: true
    schema: 'typedefs/**/*.graphql'
    documents: null
    generates:
      src/generated/graphql.ts:
        config:
          typesPrefix: I
          namingConvention:
            enumValues: change-case#constantCase
          enumValues:
            MyEnum: ../types#MyEnum
          mappers:
           ...(skip)...
        plugins:
          - 'typescript'
          - 'typescript-resolvers'
```

2. example.graphql

Please ignore why I put the enum in login input. It's just for an example :)

- Bad situation

```graphql
type Mutation {
  # situation 1
  login(
    username: String!,
    password: String!,
    myEnum: MyEnum
  ): LoginPayload
}
```

- Good situation

```graphql
input LoginInput {
  username: String! 
  password: String!
  myEnum: MyEnum
}

type Mutation {
 # situation 2
  login(input: LoginInput): LoginPayload
}
```

2. generated/graphql.ts

- Bad situation

```typescript
import { MyEnum } from '../types';

export type IMutationLoginArgs = {
  username: Scalars['String']
  password: Scalars['String']
  myEnum?: Maybe<IMyEnum> // 'Cannot find name IMyEnum' !!!
}
```

- Good situation

```typescript
import { MyEnum } from '../types';

export type ILoginInput = {
   __typename?: 'LoginInput',
  username: Scalars['String'],
  password: Scalars['String'],
  myEnum?: Maybe<MyEnum>,
};
```

**Expected behavior**

when I put enum registered as enumValues in codegen.yml in input fields, I expected the type name doesn't get prefixed in generated code.

**Environment:**

- OS: MAC
- NodeJS: v12.10.0
- package.json
```
"@graphql-codegen/cli": "^1.8.3",
"@graphql-codegen/typescript": "^1.8.3",
"@graphql-codegen/typescript-resolvers": "^1.8.3",
"typescript": "^3.7.2"
```

**Comment by dotansimha** [comment]
@roseline124 I think I managed to fix that in this PR: https://github.com/dotansimha/graphql-code-generator/pull/3015

**Comment by dotansimha** [comment]
Fixed in 1.9.0

**Comment by roseline124** [comment]
Thanks!! :)
